tests/pytests/unit/utils/parsers/test_log_parsers.py

```diff
@@ -0,0 +1,775 @@
+    return str(tmp_path / logfile_config_setting_name)
+
+
+@pytest.fixture
+def args(log_cli_parser):
+    if log_cli_parser in ("saltcmd", "saltssh"):
+        return ["foo", "bar.baz"]
+    elif log_cli_parser == "saltcp":
+        return ["foo", "bar", "baz"]
+    elif log_cli_parser in ("saltcall", "saltrun"):
+        return ["foo.bar"]
+    elif log_cli_parser == "saltcloud":
+        return ["-p", "foo", "bar"]
+    elif log_cli_parser == "spm":
+        return ["foo", "bar"]
+    return []
+
+
+@pytest.fixture
+def loglevel_config_setting_name():
+    return "log_level"
+
+
+@pytest.fixture
+def logfile_config_setting_name(log_cli_parser):
+    if log_cli_parser == "syndic":
+        return "syndic_log_file"
+    elif log_cli_parser == "saltkey":
+        return "key_logfile"
+    elif log_cli_parser == "saltssh":
+        return "ssh_log_file"
+    elif log_cli_parser == "spm":
+        return "spm_logfile"
+    elif log_cli_parser == "saltapi":
+        return "api_logfile"
+    return "log_file"
+
+
+@pytest.fixture
+def logfile_loglevel_config_setting_name():
+    return "log_level_logfile"
+
+
+@pytest.fixture
+def testing_config(default_config, root_dir, logfile_config_setting_name, log_file):
+    _testing_config = default_config.copy()
+    _testing_config["root_dir"] = root_dir
+    for name in ("pki_dir", "cachedir"):
+        _testing_config[name] = name
+    _testing_config[logfile_config_setting_name] = log_file
+    return _testing_config
+
+
+@pytest.fixture(autouse=True)
+def log_impl():
+    """
+    Mock logger functions
+    """
+    _log_impl = LogImplMock()
+    mocked_functions = {}
+    for name in dir(_log_impl):
+        if name.startswith("_"):
+            continue
+        func = getattr(_log_impl, name)
+        if not callable(func):
+            continue
+        mocked_functions[name] = func
+
+    patcher = patch.multiple(salt._logging, **mocked_functions)
+    with patcher:
+        yield _log_impl
+    _log_impl._destroy()
+
+
+def test_get_log_level_cli(
+    testing_config, loglevel_config_setting_name, args, parser, config_func, log_impl
+):
+    """
+    Tests that log level match command-line specified value
+    """
+    # Set defaults
+    default_log_level = testing_config[loglevel_config_setting_name]
+
+    # Set log level in CLI
+    log_level = "critical"
+    args = ["--log-level", log_level] + args
+
+    parser = parser()
+    with patch(config_func, MagicMock(return_value=testing_config)):
+        parser.parse_args(args)
+
+    console_log_level = getattr(parser.options, loglevel_config_setting_name)
+
+    # Check console log level setting
+    assert console_log_level == log_level
+    # Check console logger log level
+    assert log_impl.log_level_console == log_level
+    assert log_impl.config[loglevel_config_setting_name] == log_level
+    assert log_impl.temp_log_level == log_level
+    # Check log file logger log level
+    assert log_impl.log_level_logfile == default_log_level
```

Comment:
```suggestion
    instance = parser()
    with patch(config_func, MagicMock(return_value=testing_config)):
        instance.parse_args(args)

    console_log_level = getattr(instance.options, loglevel_config_setting_name)

    # Check console log level setting
    assert console_log_level == log_level
    # Check console logger log level
    assert log_impl.log_level_console == log_level
    assert log_impl.config[loglevel_config_setting_name] == log_level
    assert log_impl.temp_log_level == log_level
    # Check log file logger log level
    assert log_impl.log_level_logfile == default_log_level
```
Once we start type checking, this will become a typing problem.
Also, the same kind of change for the remaining test functions that do `parser = parser()`